internal/query/gin_map_test.go

```diff
@@ -10,103 +10,189 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestContextQueryNestedMap(t *testing.T) {
+func TestContextShouldGetQueryNestedMapSuccessfulParsing(t *testing.T) {
 	var emptyQueryMap map[string]interface{}
 
 	tests := map[string]struct {
 		url            string
 		expectedResult map[string]interface{}
-		exists         bool
 	}{
-		"no searched map key in query string": {
-			url:            "?foo=bar",
+		"no query params": {
+			url:            "",
 			expectedResult: emptyQueryMap,
-			exists:         false,
 		},
-		"searched map key is not a map": {
-			url:            "?mapkey=value",
-			expectedResult: emptyQueryMap,
-			exists:         false,
+		"single query param": {
+			url: "?foo=bar",
+			expectedResult: map[string]interface{}{
+				"foo": "bar",
+			},
 		},
-		"searched map key is array": {
-			url:            "?mapkey[]=value1&mapkey[]=value2",
-			expectedResult: emptyQueryMap,
-			exists:         false,
+		"multiple query param": {
+			url: "?foo=bar&mapkey=value1",
+			expectedResult: map[string]interface{}{
+				"foo":    "bar",
+				"mapkey": "value1",
+			},
 		},
-		"searched map key with invalid map access": {
-			url:            "?mapkey[key]nested=value",
-			expectedResult: emptyQueryMap,
-			exists:         false,
+		"map query param": {
+			url: "?mapkey[key]=value",
+			expectedResult: map[string]interface{}{
+				"mapkey": map[string]interface{}{
+					"key": "value",
+				},
+			},
 		},
-		"searched map key with valid and invalid map access": {
-			url: "?mapkey[key]invalidNested=value&mapkey[key][nested]=value1",
+		"nested map query param": {
+			url: "?mapkey[key][nested][moreNested]=value",
 			expectedResult: map[string]interface{}{
-				"key": map[string]interface{}{
-					"nested": "value1",
+				"mapkey": map[string]interface{}{
+					"key": map[string]interface{}{
+						"nested": map[string]interface{}{
+							"moreNested": "value",
+						},
+					},
 				},
 			},
-			exists: true,
 		},
-		"searched map key with valid before invalid map access": {
-			url: "?mapkey[key][nested]=value1&mapkey[key]invalidNested=value",
+		"map query param with explicit arrays accessors ([]) at the value level will return array": {
+			url: "?mapkey[key][]=value1&mapkey[key][]=value2",
 			expectedResult: map[string]interface{}{
-				"key": map[string]interface{}{
-					"nested": "value1",
+				"mapkey": map[string]interface{}{
+					"key": []string{"value1", "value2"},
+				},
+			},
+		},
+		"map query param with implicit arrays (duplicated key) at the value level will return only first value": {
+			url: "?mapkey[key]=value1&mapkey[key]=value2",
+			expectedResult: map[string]interface{}{
+				"mapkey": map[string]interface{}{
```

Comment:
Minor but general.
In relation to our team discussion - why don't use `any` instead of `interface{}`?
- for me, it would be more readable to have `map[string]any`